dd-trace-core/src/main/java/datadog/trace/core/PendingTraceBuffer.java

```diff
@@ -180,13 +212,20 @@ public void run() {
               pendingTrace = queue.take(); // block until available;
             }
 
-            if (pendingTrace instanceof FlushElement) {
+            if (pendingTrace == FLUSH_ELEMENT) {
               // Since this is an MPSC queue, the drain needs to be called on the consumer thread
               queue.drain(WriteDrain.WRITE_DRAIN);
               flushCounter.incrementAndGet();
               continue;
             }
 
+            if (pendingTrace == DUMP_ELEMENT) {
+              queue.drain(DumpDrain.DUMP_DRAIN, 50);
+              queue.fill(DumpDrain.DUMP_DRAIN, DumpDrain.DATA.size());
```

Comment:
```suggestion
              queue.fill(DumpDrain.DUMP_DRAIN, queue.drain(DumpDrain.DUMP_DRAIN, 50));
```
`queue.drain` will return the number of elements actually drained - you can use this to avoid exposing the internal collection from `DumpDrain`